crates/services/p2p/src/service.rs

```diff
@@ -131,9 +159,9 @@ impl<D> Task<D> {
         db: Arc<D>,
         block_importer: Arc<B>,
     ) -> Self {
-        let (request_sender, request_receiver) = mpsc::channel(100);
-        let (tx_broadcast, _) = broadcast::channel(100);
-        let (block_height_broadcast, _) = broadcast::channel(100);
```

Comment:
interesting, we should probably compute these channel limits based on the sync parameters for batch size and number of concurrent batches etc.